src/routes/apis/registry/list/repositories/+server.ts

```diff
@@ -0,0 +1,18 @@
+import type { RequestHandler } from './$types';
+import { env } from '$env/dynamic/public';
+import { json } from '@sveltejs/kit';
+
+export const GET: RequestHandler = async ({ fetch, url }) => {
+	const query = url.searchParams.get('query');
+	const backendUrl = new URL(
+		`/v2/ext/catalog/search?search_query=${query}`,
+		env.PUBLIC_OPEN_REGISTRY_BACKEND_URL
+	);
+
+	const response = await fetch(backendUrl);
+	if (!query) {
```

Comment:
@guacamole we should check for status code here and then return an error response if it's not a code we expect.

The query validation should be done at the top, right after we assign the query value